app/page.tsx

```diff
@@ -1,15 +1,32 @@
 "use client";
 
+import { auth0UserContext } from "@/types/userContext";
 import { useUser, withPageAuthRequired } from "@auth0/nextjs-auth0/client";
+import Layout from "./components/Layout";
+import Deliveries from "./components/Deliveries";
+import { dataDevlivries } from "../data";
+import { useContext, useEffect } from "react";
+import { DeliveryContext } from "./components/Delivery/deliveryProvider";
+import DeliveryInfo from "./components/DeliveryInfo";
 
 const Home = () => {
-  const { user } = useUser();
+  const { deliveries, handleDeliveries } = useContext(DeliveryContext);
+  const { user } = useUser() as auth0UserContext;
+
+  useEffect(() => {
+    handleDeliveries(dataDevlivries);
```

Comment:
I think you can write a memoized value that would be the transformation of "deliveries", and expose it in your context. This way, the component would consume it directly. Alternatively, you can place the memoized value here and use the parse function directly.

I think the second option is better since I'm not sure who would be interested in the transformed deliveries object other than the Deliveries page.

Reply:
Solved